### ISSUE TITLE
[Bug/Feature] Resolver to `.yarnrc` for npm/git registry

### ISSUE THREAD

**Issue by rtritto** [yarnpkg/berry#3111]
**Describe the user story**
Allow to use, in `.yarnrc.yml` file, a package version from `npm` or `git` registry.

**Example**
In this example, I used [Recoil](https://github.com/facebookexperimental/Recoil) (with `React` peer dependency), [Preact](https://github.com/preactjs/preact) (light alternative to `React`) and [React](https://github.com/facebook/react) (peer dependency of `Recoil`).
I run the install command:
`yarn add recoil preact react@npm:@preact/compat@*`
At the end of it, I get the warning for missing `react` dependency of `recoil`:
```
➤ YN0000: ┌ Resolution step
➤ YN0060: │ project@workspace:. provides react (p7c5af) with version 0.0.4, which doesn't satisfy what recoil requests
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
```

**Describe the solution you'd like**
In `.yarnrc.yml` file to `recoil` package, I added the `react` (`preact` version from `npm` registry) dependency:

```yml
packageExtensions:
  recoil@*:
    dependencies:
      react: "npm:@preact/compat@*"
```

### Actual
After I run `yarn` (to install `react` using `.yarnrc.yml` file), the warning still remains (dependency in `.yarnrc.yml` file is ignored).
### Expected
- No warning.
- `react` dependency is installed by `.yarnrc.yml` file

**Comment by rtritto** [comment]
I tested with `yarn` versions from 2.4.1 to 3.0.2 (currently latest).
Seems that this isn't yet reproducible. :thinking:

**Comment by merceyz** [comment]
It always worked, the warning was because the version of `@preact/compat` didn't match what `recoil` wanted. `@preact/compat` is now released as `17.0.2` making it match the range and thus removing the warning
